corehq/apps/hqcase/api/get_list.py

```diff
@@ -93,7 +92,15 @@ def get_list(domain, params):
         else:
             raise UserError(f"'{key}' is not a valid parameter.")
 
-    return [serialize_es_case(case) for case in query.run().hits]
+    es_result = query.run()
+    ret = {
+        "total": es_result.total,
+        "cases": [serialize_es_case(case) for case in es_result.hits],
+    }
+    cases_in_result = len(es_result.hits)
+    if cases_in_result and es_result.total > cases_in_result:
+        ret['next'] = {**raw_params, **{'indexed_on.gt': es_result.hits[-1]["@indexed_on"]}}
+    return ret
```

Comment:
This is a pretty unsophisticated approach - I just give them a URL with the same params as they used for the current request, with the addition of a new `indexed_on.gt` filter.  We could certainly b64 encode all this, though I don't think that buys us anything useful, as this mechanism doesn't need to be hidden.  The one argument I could think of is that by obscuring the means of pagination, we're less likely to break someone's integration if we changed the pagination scheme in the future.

It seems unlikely enough that someone would inspect and modify the `next` value, so I didn't bother, but I'm open to being persuaded.

Reply:
> It seems unlikely enough that someone would inspect and modify the `next` value

Even if they did, that wouldn't be a problem, would it?

Reply:
Not really, unless they did it programmatically and we wanted to change the way we do pagination without disrupting their workflow.  IDK, seems like if that did arise, it'd be on them.